salt/config/__init__.py

```diff
@@ -2190,6 +2191,7 @@ def minion_config(path,
     opts = apply_minion_config(overrides, defaults,
                                cache_minion_id=cache_minion_id,
                                minion_id=minion_id)
+    opts['__role'] = role
```

Comment:
This should already be set by the time you need it. It's already being used elsewhere in the codebase already. Are you sure you aren't seeing it?

Reply:
I believe it might be already applied, but it is `minion` even when `master` is passed through to the minion_config, from the MasterMinion instance.

I could switch this to pass `role` through, all the way to loading the sdb modules, so that vault can check that instead of `__role` in the `__opts__` object.  But since we already have it here, it seems useful to know it is a master_minion instead of just a regular minion.